File: backend/app/models/patient_qa.py
"""
models/patient_qa.py
-----------------------
Module 3: AI Patient History Q&A (README Features table - "Doctor ...
uses AI to analyze reports, answers patient-history questions"). A
doctor-facing tool, not patient/family-facing, same split as the
existing doctor-only AI Summary Email feature (routers/ai_summary.py).

Persisted (not a throwaway call) so a doctor's past Q&A for a patient is
a reusable record - same reasoning SymptomLog/DietPlan already use.
"""

import uuid
from datetime import datetime

from sqlalchemy import Column, DateTime, ForeignKey, Text
from sqlalchemy.dialects.postgresql import UUID

from app.database import Base


class PatientQuestion(Base):
    __tablename__ = "patient_questions"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    patient_id = Column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=False)
    doctor_id = Column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=False)
    question = Column(Text, nullable=False)
    answer = Column(Text, nullable=False)
    created_at = Column(DateTime, default=datetime.utcnow)

File: frontend/lib/api/patientQa.ts
/**
 * lib/api/patientQa.ts
 * -----------------------
 * Module 3: AI Patient History Q&A - doctor-only.
 */

import { apiFetch } from "@/lib/apiClient";

export interface PatientQuestionOut {
  id: string;
  patient_id: string;
  doctor_id: string;
  question: string;
  answer: string;
  created_at: string;
}

export function askPatientQuestion(patientId: string, question: string) {
  return apiFetch(`/patient-qa/${patientId}`, {
    method: "POST",
    body: JSON.stringify({ question }),
  }) as Promise<PatientQuestionOut>;
}

export function getPatientQuestionHistory(patientId: string) {
  return apiFetch(`/patient-qa/${patientId}`) as Promise<PatientQuestionOut[]>;
}
